qiskit/transpiler/synthesis/aqc/__init__.py

```diff
@@ -10,169 +10,54 @@
-
-One can take advantage of accelerated version of objective function. It implements the same
-mathematical algorithm as the default one ``DefaultCNOTUnitObjective`` but runs several times
-faster. Instantiation of accelerated objective function class is similar to the default case:
-
-    # Create an objective that defines our optimization problem
-    approximating_objective = FastCNOTUnitObjective(num_qubits=num_qubits, cnots=cnots)
-
-The rest of the code in the above example does not change.
-
-References:
-
-    [1]: Liam Madden, Andrea Simonetto, Best Approximate Quantum Compiling Problems.
-        `arXiv:2106.05649 <https://arxiv.org/abs/2106.05649>`_
 """
 
-from .approximate import ApproximateCircuit, ApproximatingObjective
-from .aqc import AQC
-from .aqc_plugin import AQCSynthesisPlugin
-from .cnot_structures import make_cnot_network
-from .cnot_unit_circuit import CNOTUnitCircuit
-from .cnot_unit_objective import CNOTUnitObjective, DefaultCNOTUnitObjective
-from .fast_gradient.fast_gradient import FastCNOTUnitObjective
+import warnings
+
+from qiskit.synthesis.unitary.aqc.approximate import ApproximateCircuit, ApproximatingObjective
+from qiskit.synthesis.unitary.aqc import AQC
+from qiskit.synthesis.unitary.aqc.cnot_structures import make_cnot_network
+from qiskit.synthesis.unitary.aqc.cnot_unit_circuit import CNOTUnitCircuit
+from qiskit.synthesis.unitary.aqc.cnot_unit_objective import (
+    CNOTUnitObjective,
+    DefaultCNOTUnitObjective,
+)
+from qiskit.synthesis.unitary.aqc.fast_gradient.fast_gradient import FastCNOTUnitObjective
+from qiskit.transpiler.passes.synthesis.aqc_plugin import AQCSynthesisPlugin
+
+warnings.warn(
+    "The qiskit.transpiler.synthesis.aqc module is pending deprecation since Qiskit 0.46.0. "
+    "It will be deprecated in a following release, no sooner than 3 months after the 0.46.0 release.",
+    stacklevel=2,
+    category=PendingDeprecationWarning,
+)
+
+_DEPRECATED_NAMES = {
+    "AQC": "qiskit.synthesis.unitary.aqc",
+    "ApproximateCircuit": "qiskit.synthesis.unitary.aqc",
+    "ApproximatingObjective": "qiskit.synthesis.unitary.aqc",
+    "CNOTUnitCircuit": "qiskit.synthesis.unitary.aqc",
+    "CNOTUnitObjective": "qiskit.synthesis.unitary.aqc",
+    "DefaultCNOTUnitObjective": "qiskit.synthesis.unitary.aqc",
+    "FastCNOTUnitObjective": "qiskit.synthesis.unitary.aqc",
+    "AQCSynthesisPlugin": "qiskit.transpiler.passes.synthesis",
+}
+
+
+def __getattr__(name):
+    if name in _DEPRECATED_NAMES:
+        import importlib
+
+        module_name = _DEPRECATED_NAMES[name]
+        warnings.warn(
+            f"Accessing '{name}' from '{__name__}' is deprecated since Qiskit 0.46"
+            f" and will be removed in 1.0.  Import from '{module_name}' instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return getattr(importlib.import_module(module_name), name)
+    raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
```

Comment:
We only need this if we deprecate some objects inside an `__init__` file and do not deprecate others. Since we here have a module-level deprecation we can remove these 🙂